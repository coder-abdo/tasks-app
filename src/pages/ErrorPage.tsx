import React from "react";
import { ErrorContainer } from "../styles/ErrorPageStyle";
export const ErrorPage = () => {
  return (
    <ErrorContainer>
      <h1>
        <span className="large">404</span>
        Ooops you lost
      </h1>
    </ErrorContainer>
  );
};
