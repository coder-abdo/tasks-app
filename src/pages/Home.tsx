import React, { useEffect } from "react";
import { useSelector, useDispatch } from "react-redux";
import { fetchingData, REMOVE_TASK } from "../actions/Action";
import Task from "../components/Task";
export const Home = () => {
  const tasks = useSelector((state: any) => state.tasks);
  const user = useSelector((state: any) => state.user);

  const dispatch = useDispatch();
  const getData = () => dispatch(fetchingData());
  useEffect(() => {
    getData();
  }, []);
  const handleRemoveTask = task =>
    dispatch({ type: REMOVE_TASK, payload: task });
  return (
    <div>
      {tasks.map((task: any) => (
        <Task
          key={task.id}
          {...task}
          handleRemoveTask={() => handleRemoveTask(task.id)}
        />
      ))}
    </div>
  );
};
