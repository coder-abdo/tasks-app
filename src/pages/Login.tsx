import React, { useState } from "react";
import { useSelector, useDispatch } from "react-redux";
import { useHistory } from "react-router-dom";
import { AUTHORIZED_USER } from "../actions/Action";
import { LoginContainer, LoginForm } from "../styles/LoginStyle";
import { IState } from "../interfaces/interfaces";

export const Login = () => {
  const userInfo = useSelector((state: IState) => state.user);
  const dispatch = useDispatch();
  const [username, setUsername] = useState<string>("");
  const [password, setPassword] = useState<string>("");
  const [err, setErr] = useState<string>("");
  const history = useHistory();

  const handleSubmit = (e: React.FormEvent) => {
    e.preventDefault();

    if (username === userInfo.username && password === userInfo.password) {
      dispatch({ type: AUTHORIZED_USER });
      history.push("/");
    } else {
      if (username !== userInfo.username) {
        setErr("username does not matched");
      } else if (password !== userInfo.password) {
        setErr("password does not matched");
      } else {
        setErr("your password or username does not matched");
      }
    }
    setUsername("");
    setPassword("");
  };
  return (
    <LoginContainer>
      <h2>hello welcome back</h2>
      <LoginForm onSubmit={handleSubmit}>
        <input
          type="text"
          name="user"
          id="user"
          placeholder="Username"
          onChange={(e: React.ChangeEvent<HTMLInputElement>) =>
            setUsername(e.target.value)
          }
          value={username}
          required
        />
        <input
          type="password"
          name="password"
          id="pass"
          placeholder="Password"
          value={password}
          onChange={(e: React.ChangeEvent<HTMLInputElement>) =>
            setPassword(e.target.value)
          }
          required
        />
        <button type="submit">Login</button>
      </LoginForm>
      {err && <div className="err-message">{err}</div>}
    </LoginContainer>
  );
};
