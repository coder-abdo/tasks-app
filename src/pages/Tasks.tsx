import React, { useState } from "react";
import { useSelector, useDispatch } from "react-redux";
import { ADD_TASK, REMOVE_TASK } from "../actions/Action";
import AppTasks from "../components/Tasks";
import Task from "../components/Task";
import TaskForm from "../styles/TasksStyle";
import { IState, ITask } from "../interfaces/interfaces";
export const Tasks = () => {
  const tasks: ITask[] = useSelector((state: IState) => state.tasks);
  const dispatch = useDispatch();
  const [title, setTitle] = useState("");
  const [cateogry, setCateogry] = useState("");
  const validateForm = () => title.length && cateogry.length;
  const handleSubmit = (e: React.FormEvent) => {
    e.preventDefault();
    if (validateForm()) {
      dispatch({
        type: ADD_TASK,
        payload: {
          title,
          cateogry
        }
      });
    } else {
      alert("error");
    }

    setTitle("");
    setTitle("");
  };

  return (
    <div>
      <h2>control your tasks</h2>
      <TaskForm onSubmit={handleSubmit}>
        <input
          type="text"
          value={title}
          onChange={(e: React.ChangeEvent<HTMLInputElement>) =>
            setTitle(e.target.value)
          }
          placeholder="task title"
          required
        />
        <input
          type="text"
          value={cateogry}
          onChange={(e: React.ChangeEvent<HTMLInputElement>) =>
            setCateogry(e.target.value)
          }
          placeholder="task category"
          required
        />
        <button type="submit">Add Task</button>
      </TaskForm>
      <AppTasks>
        {tasks.map((task: ITask) => (
          <Task
            key={task.id}
            {...task}
            handleRemoveTask={() =>
              dispatch({ type: REMOVE_TASK, payload: task.id })
            }
          />
        ))}
      </AppTasks>
    </div>
  );
};
