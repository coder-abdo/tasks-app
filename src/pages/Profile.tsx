import React from "react";
import { useSelector, useDispatch } from "react-redux";
import { FaRegUserCircle } from "react-icons/fa";
import { REMOVE_TASK } from "../actions/Action";
import Task from "../components/Task";
import Tasks from "../components/Tasks";
import { ProfileInfo } from "../styles/ProfileStyle";
import { IState, ITask } from "../interfaces/interfaces";
export const Profile = () => {
  const tasks: ITask[] = useSelector((state: IState) => state.tasks);
  const dispatch = useDispatch();
  return (
    <>
      <ProfileInfo>
        <FaRegUserCircle />
        <span>Admin </span>
        <span className="tasks-num">tasks: {tasks.length}</span>
      </ProfileInfo>
      <Tasks>
        {tasks.map((task: ITask) => (
          <Task
            key={task.id}
            {...task}
            handleRemoveTask={() =>
              dispatch({ type: REMOVE_TASK, payload: task.id })
            }
          />
        ))}
      </Tasks>
    </>
  );
};
