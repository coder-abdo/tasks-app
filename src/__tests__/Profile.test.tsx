import React from "react";
import { render, cleanup } from "@testing-library/react";
import { Profile } from "../pages/Profile";
import { Provider } from "react-redux";
import Store from "../store/store";
import { BrowserRouter as Router } from "react-router-dom";
beforeEach(cleanup);
describe("render Profile page", () => {
  it("should render Profile page properly ", () => {
    const ProfilePage = render(
      <Provider store={Store}>
        <Router>
          <Profile />
        </Router>
      </Provider>
    );
    expect(ProfilePage).toBeTruthy();
    expect(ProfilePage).toMatchSnapshot();
  });
  it("should show the number of tasks properly", () => {
    const { getByTestId } = render(
      <Provider store={Store}>
        <Router>
          <Profile />
        </Router>
      </Provider>
    );
    expect(getByTestId("tasks-numbers").textContent).toEqual("tasks: 0");
  });
});
