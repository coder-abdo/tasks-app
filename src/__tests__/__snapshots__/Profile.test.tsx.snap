// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Profile page should render Profile page properly  1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-htpNat kQAeSC"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 496 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M248 104c-53 0-96 43-96 96s43 96 96 96 96-43 96-96-43-96-96-96zm0 144c-26.5 0-48-21.5-48-48s21.5-48 48-48 48 21.5 48 48-21.5 48-48 48zm0-240C111 8 0 119 0 256s111 248 248 248 248-111 248-248S385 8 248 8zm0 448c-49.7 0-95.1-18.3-130.1-48.4 14.9-23 40.4-38.6 69.6-39.5 20.8 6.4 40.6 9.6 60.5 9.6s39.7-3.1 60.5-9.6c29.2 1 54.7 16.5 69.6 39.5-35 30.1-80.4 48.4-130.1 48.4zm162.7-84.1c-24.4-31.4-62.1-51.9-105.1-51.9-10.2 0-26 9.6-57.6 9.6-31.5 0-47.4-9.6-57.6-9.6-42.9 0-80.6 20.5-105.1 51.9C61.9 339.2 48 299.2 48 256c0-110.3 89.7-200 200-200s200 89.7 200 200c0 43.2-13.9 83.2-37.3 115.9z"
          />
        </svg>
        <span>
          Admin 
        </span>
        <span
          class="tasks-num"
          data-testid="tasks-numbers"
        >
          tasks: 
          0
        </span>
      </div>
      <div
        class="sc-bwzfXH cXhmWe"
      />
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-htpNat kQAeSC"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 496 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M248 104c-53 0-96 43-96 96s43 96 96 96 96-43 96-96-43-96-96-96zm0 144c-26.5 0-48-21.5-48-48s21.5-48 48-48 48 21.5 48 48-21.5 48-48 48zm0-240C111 8 0 119 0 256s111 248 248 248 248-111 248-248S385 8 248 8zm0 448c-49.7 0-95.1-18.3-130.1-48.4 14.9-23 40.4-38.6 69.6-39.5 20.8 6.4 40.6 9.6 60.5 9.6s39.7-3.1 60.5-9.6c29.2 1 54.7 16.5 69.6 39.5-35 30.1-80.4 48.4-130.1 48.4zm162.7-84.1c-24.4-31.4-62.1-51.9-105.1-51.9-10.2 0-26 9.6-57.6 9.6-31.5 0-47.4-9.6-57.6-9.6-42.9 0-80.6 20.5-105.1 51.9C61.9 339.2 48 299.2 48 256c0-110.3 89.7-200 200-200s200 89.7 200 200c0 43.2-13.9 83.2-37.3 115.9z"
        />
      </svg>
      <span>
        Admin 
      </span>
      <span
        class="tasks-num"
        data-testid="tasks-numbers"
      >
        tasks: 
        0
      </span>
    </div>
    <div
      class="sc-bwzfXH cXhmWe"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
